homeassistant/components/mqtt/config_flow.py

```diff
@@ -183,6 +193,54 @@ async def async_step_user(
 
         return await self.async_step_broker()
 
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Handle re-authentication with Aladdin Connect."""
+
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Confirm re-authentication with MQTT broker."""
+        errors: dict[str, str] = {}
+
+        assert self.entry is not None
+        if user_input:
+            password_changed = (user_password := user_input.get(CONF_PASSWORD)) == (
+                entry_password := self.entry.data.get(CONF_PASSWORD)
+            )
+            password = user_password if password_changed else entry_password
+            new_entry_data = {
+                **self.entry.data,
+                CONF_USERNAME: user_input.get(CONF_USERNAME),
+                CONF_PASSWORD: password,
+            }
+            if await self.hass.async_add_executor_job(
+                try_connection,
+                new_entry_data,
+            ):
+                return self.async_update_reload_and_abort(
+                    self.entry, data=new_entry_data
+                )
+
+            errors["base"] = "invalid_auth"
+
+        schema = self.add_suggested_values_to_schema(
+            REAUTH_SCHEMA,
+            {
+                CONF_USERNAME: self.entry.data.get(CONF_USERNAME),
+                CONF_PASSWORD: PWD_NOT_CHANGED,
```

Comment:
```suggestion
                CONF_PASSWORD: PWD_NOT_CHANGED,
```

Do we need to pass this at all?  I'm honestly not sure

Reply:
The option flow also allows to update the password and also sends the password to the UI.
We could use a similar flow there. We do not want users to re-add their password if it was not changed.

Reply:
I wish we could use a better sentinel but I guess we don't have a way of telling the difference between no password and empty password.

Reply:
Users can show their password, so it should be something understandable

Reply:
I don't have a better suggestion at this time. This one is easier enough to change in a future PR if someone does however